src/main/java/school/hei/haapi/service/MpbsVerificationService.java

```diff
@@ -90,14 +103,20 @@ public List<TransactionDetails> fetchThenSaveTransactionDetailsDaily() {
 
   private MpbsStatus defineMpbsStatusByOrangeStatusOrByInstantValidity(
       TransactionDetails storedTransaction, Mpbs mpbs, Instant toCompare) {
-    if (SUCCESS.equals(storedTransaction.getStatus())) {
-      return SUCCESS;
-    }
     long dayValidity = mpbs.getCreationDatetime().until(toCompare, ChronoUnit.DAYS);
 
+    if (storedTransaction == null) {
+      if (dayValidity > 2) {
+        return FAILED;
+      }
+    }
     if (dayValidity > 2) {
       return FAILED;
     }
+    if (SUCCESS.equals(storedTransaction.getStatus())) {
+      return SUCCESS;
+    }
```

Comment:
doesn't the following line resolve all of your problems ?
`SUCCESS.equals(storedTransaction.getStatus()? SUCCESS : FAILURE`

also, consider changing the name to what i previously suggested ( FAILED -> FAILURE or SUCCESS -> SUCCEEDED)